js/collaborator-agent/src/agent/manager.ts

```diff
@@ -0,0 +1,292 @@
+import { getConversationParticipantsFromAPI } from '../capabilities/actionItems';
+import { CapabilityRouter } from './router';
+
+// Result interface for manager responses
+export interface ManagerResult {
+    response: string;
+    delegatedCapability: string | null; // 'summarizer', 'action_items', 'search', or null for direct response
+    citations?: CitationAppearance[]; // Optional citations for search results
+}
+
+// Manager prompt that coordinates all sub-tasks
+export class ManagerPrompt {
+    private prompt: ChatPrompt;
+    private storage: SqliteKVStore;
+    private currentAPI?: any;
+    private currentUserId?: string;
+    private currentUserName?: string;
+    private currentUserTimezone?: string;
+    private lastDelegatedCapability: string | null = null;
+    private lastSearchCitations: CitationAppearance[] = [];
+    private router: CapabilityRouter;
+
+    constructor(storage: SqliteKVStore) {
+        this.storage = storage;
+        this.router = new CapabilityRouter();
+        this.prompt = this.initializePrompt();
+    }
+
+    private initializePrompt(): ChatPrompt {
+        const managerModelConfig = getModelConfig('manager');
+
+        const prompt = new ChatPrompt({
+            instructions: MANAGER_PROMPT,
+            model: new OpenAIChatModel({
+                model: managerModelConfig.model,
+                apiKey: managerModelConfig.apiKey,
+                endpoint: managerModelConfig.endpoint,
+                apiVersion: managerModelConfig.apiVersion,
+            }),
+        })
+            .function('delegate_to_summarizer', 'Delegate conversation analysis, summarization, or message retrieval tasks to the Summarizer Capability', {
+                type: 'object',
+                properties: {
+                    user_request: {
+                        type: 'string',
+                        description: 'The original user request to be processed by the Summarizer Capability'
+                    },
+                    conversation_id: {
+                        type: 'string',
+                        description: 'The conversation ID for context'
+                    },
+                    calculated_start_time: {
+                        type: 'string',
+                        description: 'Pre-calculated start time in ISO format (optional, only if time range is specified)'
+                    },
+                    calculated_end_time: {
+                        type: 'string',
+                        description: 'Pre-calculated end time in ISO format (optional, only if time range is specified)'
+                    },
+                    timespan_description: {
+                        type: 'string',
+                        description: 'Human-readable description of the calculated time range (optional)'
+                    }
+                },
+                required: ['user_request', 'conversation_id']
+            }, async (args: any) => {
```

Comment:
NIT: instead of "delete_to_summarizer", did you try out multiple functions for each capability? Right now you're treating all capabilities with the same exact requirement, but some capabilities may have other requirements. So what i mean is, multiple `.function` for each capabilities.

it would result in:
```
for (const capability of capabilities) {
  prompt.function(capability.name, capability.description, capability.params)
}
```

this keeps all the capability logic tightly inside the capability itself.